### DISCUSSION TITLE
MDX with NextJS & Tailwind

### DISCUSSION THREAD

**Discussion by orrymr** [mdx-js/mdx#2568] [Q&A] [answered]
Hey all,

I've been trying to get MDX working with NextJS + Tailwind, following [this guide](https://nextjs.org/docs/app/building-your-application/configuring/mdx).
I started a fresh NextJS project, and followed the instructions.

My first issue was that `mdx-components.tsx` didn't work when I put it in the /app directory. I had to put it one level up. 

Secondly, I changed my `page.tsx` file to `page.mdx`, and provided some markdown:

```markdown
# This is intro

- this
- is 
- a
- list
```

But the result looks pretty bad:

![meh](https://github.com/user-attachments/assets/46b9b003-c7fe-4eb8-b5d1-b3fdb4d773da)

Any idea what I'm doing wrong here?

**Comment by ChristianMurphy** [comment, marked as answer]
> My first issue was that mdx-components.tsx didn't work when I put it in the /app directory. I had to put it one level up.

That is next.js specific, raise this with them

> But the result looks pretty bad:

Yep, the tailwind defaults are atrocious.
It's a regular question https://github.com/orgs/mdx-js/discussions?discussions_q=is%3Aopen+tailwind

The answer is to use tailwind prose for better defaults https://dev.to/adamof/introduction-to-the-prose-class-that-instantly-beautifies-articles-tailwindcss--bn7

**Comment by orrymr** [comment]
Aha - so it seems MDX simply creates your `<h1>, <h2>, <ul>,` etc.. elements out of the markdown.
But it does not have anything to do with the styling!

`@tailwind base`  [removes all of the styling](https://tailwindcss.com/docs/preflight).

**Comment by mcmoodoo** [comment]
Also looking to integrate `mdx` blog to a current `next js` and `tailwind` project.

Is there a working demo of someone successfully implementing such a setup? Would love to take a look even at similar setups. I no longer care which tools and libraries. Just need to get this to work. I am sure there are people successfully using such setups...

Just need pointers here... anyone?

**Reply by ChristianMurphy** [reply]
@mcmoodoo did you try the answer?
https://github.com/orgs/mdx-js/discussions/2568#discussioncomment-11580925
